### PR TITLE
Collection and Query isEmpty()

### PR DESCRIPTION
**Pull request by lorenzo**
One of the most common questions for cake 3 is how to determine if the query has any results, the same question is also often asked in stackoverflow for collection objects. That made me realize we needed a method for checking collection emptiness in a non-hacky way.

### CODE REVIEW

File: src/Collection/CollectionTrait.php [hunk 1]
```diff
@@ -532,6 +532,15 @@ public function through(callable $handler)
     }
 
     /**
+     * {@inheritDoc}
+     *
+     */
+    public function isEmpty()
+    {
+        return iterator_count($this->take(1)) === 0;
```
Comment (markstory) [inline src/Collection/CollectionTrait.php]: Does take() a result mean it won't be available for iteration?

Comment (lorenzo) [inline src/Collection/CollectionTrait.php]: no

**Comment by lorenzo** [comment]
That failing test is the parseDSN stuff is super annoying

**Comment by markstory** [comment]
@lorenzo Especially because it only fails some of the time. Is it reasonable to remove the flaky test?

**Comment by lorenzo** [comment]
I will remove it

**Comment by dereuromark** [comment]
Is there any documentation on that one yet?

**Pull request merged**